projects/plugins/jetpack/extensions/blocks/like/like.php

```diff
@@ -128,43 +130,29 @@ function render_block( $attr, $content, $block ) {
  * Helper function to determine whether the Like module has been disabled
  */
 function is_legacy_likes_disabled() {
+	$settings = new \Jetpack_Likes_Settings();
+
 	$is_wpcom                 = defined( 'IS_WPCOM' ) && IS_WPCOM;
 	$is_likes_module_inactive = ! \Jetpack::is_module_active( 'likes' );
 	$is_disabled_on_wpcom     = $is_wpcom && get_option( 'disabled_likes' ) && get_option( 'disabled_reblogs' );
 	$is_disabled_on_non_wpcom = ! $is_wpcom && get_option( 'disabled_likes' );
-
-	return $is_likes_module_inactive || $is_disabled_on_wpcom || $is_disabled_on_non_wpcom;
+	return $is_likes_module_inactive || $is_disabled_on_wpcom || $is_disabled_on_non_wpcom || ! $settings->is_likes_module_enabled();
 }
 
 /**
- * Registers and enqueues script and style for Jetpack Likes.
- *
- * This function conditionally registers and enqueues the Jetpack Likes
- * CSS and JavaScript files based on the environment. It handles both
- * WordPress.com and self-hosted WordPress environments. The function
- * relies on `IS_WPCOM` to determine the environment and uses different
- * paths for script and style files accordingly.
- *
- * @return void Early return if legacy likes are not disabled.
+ * Renders the iframe and enqueues the necessary scripts.
  */
-function register_script_and_style() {
-	if ( ! is_legacy_likes_disabled() ) {
-		return;
-	}
+function render_iframe() {
+	static $main_iframe_added = false;
 
-	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-		$style_url  = content_url( 'mu-plugins/likes/jetpack-likes.css' );
-		$script_url = content_url( 'mu-plugins/likes/queuehandler.js' );
-	} else {
-		$style_url  = plugins_url( 'modules/likes/style.css', dirname( __DIR__, 2 ) );
-		$script_url = Assets::get_file_url_for_environment(
-			'_inc/build/likes/queuehandler.min.js',
-			'modules/likes/queuehandler.js'
-		);
-	}
+	if ( ! $main_iframe_added ) {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			Jetpack_Likes::likes_master();
```

Comment:
We are getting a Fatal on this line:

```
Fatal error: Uncaught Error: Class 'Automattic\Jetpack\Extensions\Like\Jetpack_Likes' not found in /home/wpcom/public_html/wp-content/mu-plugins/jetpack-plugin/sun/extensions/blocks/like/like.php:150
```

The following change will do the trick:

```suggestion
			\Jetpack_Likes::likes_master();
```